source/core/hash.h

```diff
@@ -43,7 +43,18 @@ namespace Slang
         }
         return hash;
     }
-    
+
+    inline uint64_t GetHashCode64(const char * buffer, size_t numChars)
```

Comment:
I think the basecode that Yong started the project with had an MD5 hash implementation. We should probably aspire to use something documented and off-the-shelf for a semi-critical hash like this (especially if we are going to be treating collisions as impossible; I guess I'll see whether you fall back to actually comparing the strings...).

Reply:
Agreed we could use a better hash. I did look through source, and only found this one. I also did look around for good hashing function - but the best rated (speed/hits) are complex and will probably need to be an external dependency and may have licensing issues.